zipline/data/minute_bars.py

```diff
@@ -391,15 +396,16 @@ def __init__(self,
         self._minute_index = _calc_minute_index(
             self._schedule.market_open, self._minutes_per_day)
 
-        metadata = BcolzMinuteBarMetadata(
-            self._default_ohlc_ratio,
-            self._ohlc_ratios_per_sid,
-            self._calendar,
-            self._start_session,
-            self._end_session,
-            self._minutes_per_day,
-        )
-        metadata.write(self._rootdir)
+        if write_metadata:
+            metadata = BcolzMinuteBarMetadata(
+                self._default_ohlc_ratio,
+                self._ohlc_ratios_per_sid,
+                self._calendar,
+                self._start_session,
+                self._end_session,
```

Comment:
In the `truncate` case `_end_session` will need to be updated. Does `truncate` make the metadata change?